Clamp GetMaxSkill level to maximum size of g_skillTable

claywar opened the pull request:
<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

Pets have the ability to exceed level 99 with iLevel/gear/merits.  Loading pet will perform a check for elemental magic level of a pet, causing an invalid index being passed to `GetMaxSkill()`

Fixes #1320

TeoTwawki commented:
> Pets have the ability to exceed level 99 with iLevel/gear/merits. Loading pet will perform a check for elemental magic level of a pet, causing an invalid index being passed to GetMaxSkill()
> 
> This copies lv99 values for 30 additional levels as a measure to prevent crash, but does not implement higher values (and not sure if we should be, since player skills do not raise with item level.

prolly should have that info right in the commit message so we could always see why this was done. Github commentary can be lost, commits are forever

zach2good commented:
If the higher skills are never going to change from the 99 entry (ever), we could do this:
```cpp
    uint16 GetMaxSkill(SKILLTYPE SkillID, JOBTYPE JobID, uint8 level)
    {
        return g_SkillTable[std::clamp<uint8>(level, 0, 99)][g_SkillRanks[SkillID][JobID]];
    }
```

zach2good commented:
If the higher skills are never going to change from the 99 entry (ever), we could do this:
```cpp
    uint16 GetMaxSkill(SKILLTYPE SkillID, JOBTYPE JobID, uint8 level)
    {
        return g_SkillTable[std::clamp<uint8>(level, 0, 99)][g_SkillRanks[SkillID][JobID]];
    }
```

zach2good commented:
If the higher skills are never going to change from the 99 entry (ever), we could do this:
```cpp
    uint16 GetMaxSkill(SKILLTYPE SkillID, JOBTYPE JobID, uint8 level)
    {
        return g_SkillTable[std::clamp<uint8>(level, 0, 99)][g_SkillRanks[SkillID][JobID]];
    }
```

TeoTwawki commented:
> If the higher skills are never going to change from the 99 entry (ever), we could do this:
> 
> ```c++
>     uint16 GetMaxSkill(SKILLTYPE SkillID, JOBTYPE JobID, uint8 level)
>     {
>         return g_SkillTable[std::clamp<uint8>(level, 0, 99)][g_SkillRanks[SkillID][JobID]];
>     }
> ```

This will depend on what reatil actually does for nonj-player entities that have actual skill levels (DO they have actual skill levels? I suspect not. I believe they are just getting attack acc evasion and such directly)

zach2good commented:
If the higher skills are never going to change from the 99 entry (ever), we could do this:
```cpp
    uint16 GetMaxSkill(SKILLTYPE SkillID, JOBTYPE JobID, uint8 level)
    {
        return g_SkillTable[std::clamp<uint8>(level, 0, 99)][g_SkillRanks[SkillID][JobID]];
    }
```

claywar committed:
Clamp GetMaxSkill value to Array Size

Pets have the ability to exceed level 99 with iLevel/gear/merits. Loading pet will perform a check for elemental magic level of a pet, causing an invalid index being passed to GetMaxSkill()

Add comments and general cleanup


Add same safety to other GetMaxSkill function
